Migrate THCThrustAllocator to ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65746
* #65745
* #65744
* #65743
* __->__ #65492

Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)
BC-breaking note:
This PR removes THCThrustAllocator.cuh, if your extension is using THCThrustAllocator.cuh, replace it with ATen/cuda/ThrustAllocator.h and corresponding APIs (see examples in this PR).
This PR removes THCudaMalloc/THCudaFree calls, please use c10::cuda::CUDACachingAllocator::raw_alloc(size)/raw_delete(ptr), or, preferably, switch to c10:cuda::CUDaCachingAllocator::allocate that manages deallocation.

peterbell10 committed:
Migrate THCThrustAllocator to ATen

[ghstack-poisoned]

peterbell10 committed:
Remove ATenCeilDiv in favour of ceil_div on "Migrate THCThrustAllocator to ATen"




[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"




[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"




[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148180).

peterbell10 committed:
Remove manual hipifying include path on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148180).

ngimel commented:
ROCm build error is real, windows probably a fluke.

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148180).

ngimel commented:
rocm error still exists, looks like you need to call  c10::hip::HIPCachingAllocator directly and not rely on hipification, as on line 484

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCThrustAllocator to ATen"


Differential Revision: [D31148180](https://our.internmc.facebook.com/intern/diff/D31148180)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148180).

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148180).
